domains/certificates/Query.API/API/ContractService/CertificateIssuingContract.cs

```diff
@@ -14,6 +16,34 @@ public class CertificateIssuingContract
     public DateTimeOffset StartDate { get; set; }
     public DateTimeOffset? EndDate { get; set; }
     public DateTimeOffset Created { get; set; }
+    public string WalletUrl { get; set; } = "";
+    public byte[] WalletPublicKey { get; set; } = Array.Empty<byte>();
+
+    public IHDPublicKey GetHDPublicKey() => new Secp256k1Algorithm().ImportHDPublicKey(WalletPublicKey);
```

Comment:
It is needed later when sending the certificate to the Wallet. The problem is, that if it is not validated here upon Contract creation, the problem will show in a background process running several hours or days later.

Reply:
I would prefer storing an IHDPublicKey in the database. Figuring out how to do that does with MartenDB will take time - and since we will switch to EF Core at a later point, I see spending time on this detail in MartenDB as wasted. So I went with a solution where the validation happens in code, and not in the ORM.